src/babel/utils.ts

```diff
@@ -21,10 +21,19 @@ const atomFunctionNames = [
   'atom',
   'atomFamily',
   'atomWithDefault',
+  'atomWithHash',
+  'atomWithImmer',
+  'atomWithInfiniteQuery',
+  'atomWithMachine',
+  'atomWithMutation',
   'atomWithObservable',
+  'atomWithProxy',
+  'atomWithQuery',
   'atomWithReducer',
   'atomWithReset',
+  'atomWithSubscription',
   'atomWithStorage',
+  'atomWithStore',
   'freezeAtom',
   'loadable',
   'selectAtom',
```

Comment:
I actually wonder if we should cache `selectAtom` and `splitAtom`.
Anyway, merging this for now.